### ISSUE TITLE
Wrong suggestions for crossreference with national character

### ISSUE THREAD

**Issue by luvarqpp** [asciidoctor/asciidoctor-intellij-plugin#598]
## Observed vs. expected behavior

IDE has suggested me wrong reference. It was reference to title which had national characters in it.

## Steps to reproduce

Prepare some adoc document with this content:
```
= Nejaka titulka
Ľubomír Varga, QPP a.s. 2020
Version 0.2, 17.11.2020
:toc:

== Nadpisík

A teraz referencia na prvú kapitolu
```

Than move your caret to the end and write `<<`. This will popup list of possible references and one of them will be `_nadpisk`. There is missing "í" character.

## Environment

Plugin Version: 0.31.41

IntelliJ Details: IntelliJ IDEA 2020.2.3 (Community Edition)
Build #IC-202.7660.26, built on October 6, 2020
Runtime version: 11.0.8+10-b944.34 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-52-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 1976M
Cores: 8
Non-Bundled Plugins: PlantUML integration, DBN, Lombook Plugin, com.oliverlockwood.plugins.jenkinsfile, im.jeanfrancois.ideaplugins.ideafullscreen, training, net.seesharpsoft.intellij.plugins.csv, org.jetbrains.plugins.hocon, Docker, org.jetbrains.kotlin, org.intellij.scala, com.intellij.lang.jsgraphql, org.asciidoctor.intellij.asciidoc
Current Desktop: i3

**Comment by ahus1** [comment]
Thanks for reporting this. I missed Unicode options for two regex expressions, this will be fixed for the next pre-release. Travis CI is triggered, and now I have to wait for the build queue. As they are shutting down their old service they started to reduce their capacity. Hopefully it will get better once we move for example to GitHub actions, see #597 

I see that you're using a pre-release: Thanks for giving them a try, this is always appreciated!

**Comment by ahus1** [comment]
Pre-release 0.31.42 is available that includes this fix. Please let me know if this works for you.

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

**Comment by luvarqpp** [comment]
**Fixed**. I have 0.31.43 now and it is working as expected... You can close this issue from my point of view.

A bit offtopic question from me, regarding titles and crosslinks. Consider document like this:

```
:simulatorName: QPPElektromer


== Čo je {simulatorName} a na čo slúži?

<<_čo_je_qppelektromer_a_na_čo_slúži>>
```
Now when I change *simulatorName* variable, crossreference is underlined (as inspection "Link resolve inspection" starts to fail), but title is not "re-rendered". Here in source you see variable name in brackets, but in idea (if I have caret on other line), brackets and variable name is replaced by variable value and green-ish background. This thing is not updated when I change variable value.

PS: There is also problem, that changing variable value which is in some title, will invalidate all cross-references (implicit anchors). Perhaps some refactor functionality would be handy, if this technique is used in day-to-day work.

**Comment by ahus1** [comment]
Thanks for testing it. I close the ticket.

I can't reproduce the non-changing title or non-changing green-ish (folded) attributes, see the attached recording. The folded attribute change with a slight sub-second delay. That the reference is not changing is the way it is supported today. I recommend custom IDs to solve that (see next paragraph).

Regarding changing titles, attributes and references: Using the title as-is is as a reference always problematic regarding renaming: The reference needs to change, event if it points to the same paragraph as before. I recommend using [custom IDs](https://asciidoctor.org/docs/user-manual/#custom-ids), as they are explicit and don't change then the title changes. For custom IDs the plugin supporting renaming support (Shift+F6).

![folging](https://user-images.githubusercontent.com/3957921/100519608-5b82fb00-3199-11eb-8e44-9e38b9b5f5b5.gif)

**Comment by ahus1** [comment]
@luvarqpp - I just published pre-release 0.32.2 that fixes a problem in indexing sections with attributes. I recommend you update to this latest pre-release. 

This fixes issues if you want to search-by-name (Shift+Shift) for sections with replaced attributes in their titles.
